---
title: "austraits"
author: "Fonti Kar, Elizbeth Wenk, Daniel Falster"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{austraits}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  message = FALSE,
  warning = FALSE,
  comment = "#>",
  fig.path = "man/figures/"
)

library(dplyr)
library(stringr)
library(austraits)

path = "ignore/data/austraits/"
austraits <- load_austraits(version = "6.0.0", path = path)
```

`austraits` allow users to **access, explore and wrangle data** from [traits.build](https://github.com/traitecoevo/traits.build) relational databases. It is also an R interface to [AusTraits](https://austraits.org/), the Australian plant trait database. This package contains functions for joining data from various tables, filtering to specific records, combining multiple databases and visualising the distribution of the data. Below, we've include a tutorial using the AusTraits database to illustrate how some these functions work together to generate useful outputs.

## Install and load `austraits`
`austraits` is still under development. To install the current version from GitHub: 

```{r setup, eval = FALSE}
#install.packages("remotes")
remotes::install_github("traitecoevo/austraits", dependencies = TRUE, upgrade = "ask")

# Load the austraits package
library(austraits)
```

### Retrieve AusTraits database

We will use the latest AusTraits database as an example database. 

We can download the AusTraits database by calling `load_austraits()`. This function will download AusTraits to a specified path. By default it is `data/austraits`. The function will reload the database from this location in the future. You can set `update = TRUE` so the database is downloaded fresh from [Zenodo](https://zenodo.org/record/3568429). Note that `load_austraits()` will happily accept a DOI of a particular version.

```{r, load_data, eval = FALSE}
austraits <- load_austraits(version = "6.0.0", path = "data/austraits")
```

You can check out different versions of AusTraits and their associated DOI by using: 

```{r, eval = FALSE}
get_versions(path = "data/austraits")
```

AusTraits, like all traits.build databases, is a relational database. In R, it is a very *large* list with multiple tables. If you are not familiar with working with lists in R, we recommend having a quick look at this [tutorial](https://www.tutorialspoint.com/r/r_lists.htm). To learn more about the structure of `austraits`, check out the [structure of the database](https://traitecoevo.github.io/austraits/articles/structure.html). 

```{r}
austraits
```

## Descriptive summaries of traits and taxa

AusTraits contains `r length(unique(austraits$traits$trait_name))` plant traits. Check out [definitions of the traits](https://w3id.org/APD) to learn more about how each trait is defined. 

Have a look at data coverage by trait or taxa with: 

```{r}
summarise_austraits(austraits, "trait_name") 

summarise_austraits(austraits, "family") 

summarise_austraits(austraits, "genus") 
```

## Quickly look up data

Interested in a specific trait or context property, but unsure what terms we use? Try our `lookup_` functions.

```{r}
lookup_trait(austraits, "leaf") %>% head()
lookup_context_property(austraits, "fire") %>% head() 
lookup_location_property(austraits, "temperature") %>% head()
```

## Extracting data

In most cases, users would like to extract a subset of a database for their research purposes. 

- `extract_dataset()` filters for a particular study
- `extract_trait()` filters for a certain trait
- `extract_taxa()` filters for a specific taxon

Note you can supply a vector to each of these functions to filter for more than one study/trait/taxa. All our `extract_` function supports partial matching e.g. `extract_trait("leaf")` would return all traits containing `leaf`. 

If you would like to extract from other tables or columns, use [`extract_data`](#extract-data)

All `extract_` functions simultaneously filter across all tables in the database.

### Extracting by dataset

Filtering **one particular dataset** and assigning it to an object

```{r, extract_study}
one_study <- extract_dataset(austraits, "Falster_2005_2")

one_study$traits 
```

Filtering **multiple datasets** and assigning it to an object

```{r, extract_studies}
multi_studies <- extract_dataset(austraits, 
                                        dataset_id = c("Thompson_2001","Ilic_2000"))
 
multi_studies$traits 
```

Filtering **multiple datasets by same lead author** (e.g. Falster) and assigning it to an object.  

```{r, extract_Falster_studies}
falster_studies <- extract_dataset(austraits, "Falster")

falster_studies$traits 
```

### Extracting by taxonomy

```{r extract_taxa}
# By family 
proteaceae <- extract_taxa(austraits, family = "Proteaceae")
# Checking that only taxa in Proteaceae have been extracted
proteaceae$taxa$family %>% unique()

# By genus 
acacia <- extract_taxa(austraits, genus = "Acacia")
# Checking that only taxa in Acacia have been extracted
acacia$traits$taxon_name %>% unique() %>% head()
acacia$taxa$genus %>% unique()
```

### Extracting by trait 

```{r, extract_trait}
data_fruit <- extract_trait(austraits, "fruit")

data_fruit$traits 
```

Combining `lookup_trait` with `extract_trait` to obtain **all traits with 'leaf' in the trait name** and assigning it to an object. Note we use the `.` notation to pass on the `lookup_trait` results to `extract_trait`

```{r}
leaf <- lookup_trait(austraits, "leaf") %>% extract_trait(austraits, .) 

leaf$traits
```

### Extracting from other tables {#extract-data}

You may want to extract data from tables that have specific column values. For example calling the code below will return data where "fire" is mentioned in the `context_property` column

```{r}
data_fire <- extract_data(austraits, 
                          table =  "contexts",
                          col =  "context_property", 
                          col_value = "fire")
```

### Extracting from a single table

If you have already manipulated the original database and are working with just the traits table, the extract functions will also work on a single table.

```{r}
seedling_data <- extract_data(austraits$traits,
                          col =  "life_stage", 
                          col_value = "seedling")

Falster_data <- extract_data(trait_table,
                          col =  "dataset_id", 
                          col_value = "Falster")

leaf_data <- extract_trait(trait_table, 
                          c("leaf_area", "leaf_N_per_dry_mass"))
```

## Join data from other tables

Once users have extracted the data they want, they may want to merge other study details into the main `traits` dataframe for their analyses. For example, users may require taxonomic information for a phylogenetic analysis. This is where the `join_` functions come in. 

There are five `join_` functions in total, each designed to append specific information from other tables and elements in the `austraits` object. Their suffixes refer to the type of information that is joined, e.g. `join_taxa` appends taxonomic information to the `traits` dataframe. 

- `join_taxa()`
- `join_methods()`
- `join_location_coordinates()`
- `join_location_properties()`
- `join_context_properties()`

We recommend pulling up the help file for each one for more details e.g `?join_location_coordinates()` 

Each of the functions has specific default parameters and formatting, but offers versatile joining options.
```{r, join_}
# Join taxonomic information 
(data_fire %>% join_taxa)$traits 

# Join methodological information 
(data_fire %>% join_methods)$traits

# Join location coordinates 
(data_fire %>% join_location_coordinates)$traits 

# Join information pertaining to location properties 
(data_fire %>% join_location_properties)$traits 

# Join information pertaining to location properties 
(data_fire %>% join_location_properties(format = "many_columns", vars = "temperature, min MAT (C)"))$traits 

# Join context information 
(data_fire %>% join_context_properties)$traits

# Join information from multiple tables 
(data_fire %>% join_context_properties %>% join_location_coordinates)$traits 
```

Alternatively,users can join **all** information using `flatten_database()`: 

```{r}
data_fire %>% flatten_database() 
```

## Visualising data by site

`plot_locations()` graphically summarises where trait data was collected from and how much data is available. The legend refers to the number of neighbouring points: the warmer the colour, the more data that is available. This function only works for studies that are geo-referenced. Users must first use `join_location_coordinates()` to append latitude and longitude information from the locations dataframe into the traits dataframe before plotting.

`plot_locations()` defaults to dividing the data by trait_name (feature = "trait_name"), but you can select any of the columns within the traits table - including columns you add with `join_` functions. However, selecting `taxon_name` will likely crash R if you are working with a dataframe that still contains a large number of species.

```{r, site_plot, fig.align = "center", fig.width=5, fig.height=5}
data_fire <- data_fire %>% join_location_coordinates()
plot_locations(data_fire$traits)
```

## Visualising data distribution and variance

`plot_trait_distribution()` creates histograms and [beeswarm plots](https://github.com/eclarke/ggbeeswarm) for specific traits to help users visualise the variance of the data. Users can specify whether to create separate beeswarm plots at the level of taxonomic family, genus or by a column in the traits table, such as `dataset_id`

```{r, beeswarm, fig.align = "center", fig.width=6, fig.height=7}
austraits %>% plot_trait_distribution_beeswarm(trait_name = "wood_density", y_axis_category = "family")
austraits %>% plot_trait_distribution_beeswarm(trait_name = "wood_density", y_axis_category = "dataset_id")
```

## Reshaping the traits table

The traits table in AusTraits is in **long** format, where data for all trait information are denoted by two columns called `trait_name` and `value`. You can convert this to wide format, where each trait is in a separate column, using the function `trait_pivot_wider()`. 

Note that this function operates differently depending on the version of AusTraits. The following columns are lost when pivoting: unit, replicates, measurement_remarks, and basis_of_value.

### Pivot wider 

**Note** that the latest version of `trait_pivot_wider()` is not longer AusTraits <=4.0.2. Please refer to our [README](https://github.com/traitecoevo/austraits) to install an older version of the `austraits` R package to work old versions of the AusTraits database. 

```{r}
data_fire %>% trait_pivot_wider()
```

### Binding trait values 

Some datasets will have *multiple observations* for some traits, for instance datasets from floras often report a minimum and maximum fruit length for a species. You can use `bind_trait_values` to merge these into a single cell. 

```{r, pivot}
data_fruit <- austraits %>% 
  extract_trait("fruit_length") %>% 
  extract_taxa(family = "Rutaceae") %>% 
  extract_data(table = "traits", col = "value_type", col_value = c("minimum", "maximum"))

data_trait_bound <- data_fruit$traits %>%
  bind_trait_values() # Joining multiple obs with `--`
  
data_trait_bound  %>%   
  dplyr::filter(stringr::str_detect(value, "--"))
```

If you would like to revert the bounded trait values, you have to use `trait_pivot_longer` first, then call `separate_trait_values`: 

```{r}
data_trait_bound %>% 
  separate_trait_values(., austraits$definitions)
```

### Summarise trait values

**THIS FUNCTION IS CURRENTLY BROKEN. IT'S OUTPUT IS MISLEADING! DO NOT USE**

Alternatively, if you don't want to bind the trait values, you can give `summarise_trait_means` a try, which computes means with multiple observations. Once the  the traits table is condensed, you won't be able to revert it back.   Note that function uses an uninformed method to compute means, we recommend using it with caution. 

```{r, pivot_summarised}
data_wide_summarised <- data_fire$traits %>%
  summarise_trait_means() %>% 
  trait_pivot_wider()

data_wide_summarised$value %>% head()
```

