<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>austraits • austraits</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="austraits">
<meta property="og:description" content="austraits">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">austraits</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/austraits.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/dictionary.html">Trait definitions</a>
    </li>
    <li>
      <a href="../articles/structure.html">UNKNOWN TITLE</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/traitecoevo/austraits/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="austraits_files/header-attrs-2.11/header-attrs.js"></script><script src="austraits_files/clipboard-1.7.1/clipboard.min.js"></script><link href="austraits_files/primer-tooltips-1.4.0/build.css" rel="stylesheet">
<link href="austraits_files/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet">
<script src="austraits_files/klippy-0.0.0.9500/js/klippy.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>austraits</h1>
                        <h4 class="author">Fonti Kar, Daniel Falster</h4>
            
            <h4 class="date">2021-12-07</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/traitecoevo/austraits/blob/master/vignettes/austraits.Rmd"><code>vignettes/austraits.Rmd</code></a></small>
      <div class="hidden name"><code>austraits.Rmd</code></div>

    </div>

    
    
<script>
  addClassKlippyTo("pre.r, pre.markdown");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script><p>The <code>austraits</code> package is designed to aid users to access data from <a href="https://zenodo.org/record/5112001">AusTraits</a>, a curated plant trait database for the Australian flora. This package contains several core functions to wrangle and visualise data. Below we included a tutorial to illustrate how to use these functions in greater detail.</p>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h2>
<p><code>austraits</code> is still under development. To install the current version from GitHub</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#install.packages("remotes")</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"traitecoevo/austraits"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Load the austraits package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://traitecoevo.github.io/austraits/">austraits</a></span><span class="op">)</span></code></pre></div>
<div id="loading-austraits-database" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-austraits-database" class="anchor"></a>Loading AusTraits database</h3>
<p>By default, <code>load_austraits</code> will download <code>austraits</code> to <code>data/austraits</code> and will reload it from this location in the future. You can also specify where you want to download austraits by providing your own file path.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">austraits</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_austraits.html">load_austraits</a></span><span class="op">(</span><span class="st">"data/"</span><span class="op">)</span></code></pre></div>
<p>Note the <code>austraits</code> object is a very <em>long</em> list with various of elements. If you are not familiar with working with lists in R, we recommend having a quick look at this <a href="https://www.tutorialspoint.com/r/r_lists.htm">tutorial</a>. To learn more about the structure of <code>austraits</code>, check out the <a href="https://traitecoevo.github.io/austraits/articles/structure.html">structure of the database</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">austraits</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; [1] "list"</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The names of the different lists in `austraits`</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">austraits</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt;  [1] "traits"            "sites"             "contexts"         
#&gt;  [4] "methods"           "excluded_data"     "taxonomic_updates"
#&gt;  [7] "taxa"              "definitions"       "contributors"     
#&gt; [10] "sources"           "build_info"</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The structure of the `traits` table only</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">austraits</span><span class="op">$</span><span class="va">traits</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; tibble [905 × 12] (S3: tbl_df/tbl/data.frame)
#&gt;  $ dataset_id    : chr [1:905] "Falster_2003" "Falster_2003" "Falster_2003" "Falster_2003" ...
#&gt;  $ taxon_name    : chr [1:905] "Acacia myrtifolia" "Acacia myrtifolia" "Acacia myrtifolia" "Acacia suaveolens" ...
#&gt;  $ site_name     : chr [1:905] "Ku-ring-gai Chase National Park low nutrient" "Ku-ring-gai Chase National Park low nutrient" "Ku-ring-gai Chase National Park low nutrient" "Ku-ring-gai Chase National Park low nutrient" ...
#&gt;  $ context_name  : chr [1:905] NA NA NA NA ...
#&gt;  $ observation_id: chr [1:905] "Falster_2003_01" "Falster_2003_01" "Falster_2003_01" "Falster_2003_02" ...
#&gt;  $ trait_name    : chr [1:905] "leaf_angle" "leaf_area" "leaf_compoundness" "leaf_angle" ...
#&gt;  $ value         : chr [1:905] "66.1" "319" "simple" "71.7" ...
#&gt;  $ unit          : chr [1:905] "degrees" "mm2" NA "degrees" ...
#&gt;  $ date          : chr [1:905] NA NA NA NA ...
#&gt;  $ value_type    : Factor w/ 17 levels "raw_value","site_min",..: 3 3 9 3 3 9 3 3 9 3 ...
#&gt;  $ replicates    : chr [1:905] "3" "3" NA "3" ...
#&gt;  $ original_name : chr [1:905] "Acacia myrtifolia" "Acacia myrtifolia" "Acacia myrtifolia" "Acacia suaveolens" ...</code></pre>
</div>
</div>
<div id="extracting-data" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-data" class="anchor"></a>Extracting data</h2>
<p>AusTraits contains 11 plant traits. Check out <a href="https://traitecoevo.github.io/austraits/articles/dictionary.html">definitions of the traits</a> to learn more about how each trait is defined.</p>
<p>In most cases, users would like to extract a subset of <code>austraits</code> for their own research purposes.<code>extract_dataset</code> subsets a particular study, whereas <code>extract_trait</code>subsets by certain traits. Note that the other tables and elements of the AusTraits data are extracted too, not just the main trait table. See <code><a href="../reference/extract_dataset.html">?extract_dataset</a></code> and <code><a href="../reference/extract_trait.html">?extract_trait</a></code> for more details</p>
<div id="extracting-by-study" class="section level3">
<h3 class="hasAnchor">
<a href="#extracting-by-study" class="anchor"></a>Extracting by study</h3>
<p>Filtering <strong>one particular study</strong> and assigning it to an object</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_zanne_09</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_dataset.html">extract_dataset</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"Zanne_2009"</span><span class="op">)</span>

<span class="va">data_zanne_09</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 12
#&gt;   dataset_id taxon_name   site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;      &lt;chr&gt;        &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;
#&gt; 1 Zanne_2009 Acacia aula… &lt;NA&gt;      &lt;NA&gt;         Zanne_2009_001 wood_dens… 0.594
#&gt; 2 Zanne_2009 Adenanthera… &lt;NA&gt;      &lt;NA&gt;         Zanne_2009_002 wood_dens… 0.775
#&gt; 3 Zanne_2009 Albizia pro… &lt;NA&gt;      &lt;NA&gt;         Zanne_2009_008 wood_dens… 0.56 
#&gt; 4 Zanne_2009 Aleurites m… &lt;NA&gt;      &lt;NA&gt;         Zanne_2009_009 wood_dens… 0.327
#&gt; 5 Zanne_2009 Alphitonia … &lt;NA&gt;      &lt;NA&gt;         Zanne_2009_010 wood_dens… 0.525
#&gt; 6 Zanne_2009 Alstonia sc… &lt;NA&gt;      &lt;NA&gt;         Zanne_2009_012 wood_dens… 0.327
#&gt; # … with 5 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;</code></pre>
<p>Filtering <strong>multiple studies by two different lead authors</strong> and assigning it to an object</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_falster_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_dataset.html">extract_dataset</a></span><span class="op">(</span><span class="va">austraits</span>, 
                                        dataset_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Falster_2003"</span>,<span class="st">"Zanne_2009"</span><span class="op">)</span><span class="op">)</span>
 
<span class="va">data_falster_studies</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 12
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle 66.1 
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp… simp…
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle 71.7 
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp… simp…
#&gt; # … with 5 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;</code></pre>
<p>Filtering <strong>multiple studies by same lead author</strong> (e.g. Falster) and assigning it to an object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First, we need to identify all studies with an id that includes "Falster"</span>

<span class="op">(</span> <span class="va">dataset_ids</span> <span class="op">&lt;-</span> <span class="va">austraits</span><span class="op">$</span><span class="va">methods</span><span class="op">$</span><span class="va">dataset_id</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">.</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Falster"</span>,<span class="va">.</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<pre><code>#&gt; [1] "Falster_2003"   "Falster_2005_1" "Falster_2005_2"</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Then we extract</span>
<span class="va">data_falster_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_dataset.html">extract_dataset</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="va">dataset_ids</span><span class="op">)</span>

<span class="va">data_falster_studies</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 12
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle 66.1 
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp… simp…
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle 71.7 
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp… simp…
#&gt; # … with 5 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;</code></pre>
</div>
<div id="extracting-by-taxonomic-level" class="section level3">
<h3 class="hasAnchor">
<a href="#extracting-by-taxonomic-level" class="anchor"></a>Extracting by taxonomic level</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># By family </span>
<span class="va">proteaceae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_taxa.html">extract_taxa</a></span><span class="op">(</span><span class="va">austraits</span>, family <span class="op">=</span> <span class="st">"Proteaceae"</span><span class="op">)</span>
<span class="co"># Checking that only taxa in Proteaceae have been extracted</span>
<span class="va">proteaceae</span><span class="op">$</span><span class="va">taxa</span><span class="op">$</span><span class="va">family</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; [1] "Proteaceae"</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># By genus </span>
<span class="va">acacia</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_taxa.html">extract_taxa</a></span><span class="op">(</span><span class="va">austraits</span>, genus <span class="op">=</span> <span class="st">"Acacia"</span><span class="op">)</span>
<span class="co"># Checking that only taxa in Acacia have been extracted</span>
<span class="va">acacia</span><span class="op">$</span><span class="va">traits</span><span class="op">$</span><span class="va">taxon_name</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt;  [1] "Acacia myrtifolia"     "Acacia suaveolens"     "Acacia floribunda"    
#&gt;  [4] "Acacia celsa"          "Acacia longifolia"     "Acacia terminalis"    
#&gt;  [7] "Acacia ulicifolia"     "Acacia aulacocarpa"    "Acacia pendula"       
#&gt; [10] "Acacia mangium"        "Acacia mearnsii"       "Acacia melanoxylon"   
#&gt; [13] "Acacia pubescens"      "Acacia aneura"         "Acacia auriculiformis"
#&gt; [16] "Acacia podalyriifolia" "Acacia oraria"</code></pre>
</div>
<div id="extracting-by-trait" class="section level3">
<h3 class="hasAnchor">
<a href="#extracting-by-trait" class="anchor"></a>Extracting by trait</h3>
<p>Filtering <strong>one particular trait</strong> and assigning it to an object</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_wood_dens</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_trait.html">extract_trait</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"wood_density"</span><span class="op">)</span>

<span class="va">data_wood_dens</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 12
#&gt;   dataset_id  taxon_name  site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;dbl&gt;
#&gt; 1 Falster_20… Acacia cel… Atherton  &lt;NA&gt;         Falster_2005_… wood_dens… 0.498
#&gt; 2 Falster_20… Acronychia… Atherton  &lt;NA&gt;         Falster_2005_… wood_dens… 0.525
#&gt; 3 Falster_20… Alphitonia… Atherton  &lt;NA&gt;         Falster_2005_… wood_dens… 0.413
#&gt; 4 Falster_20… Glochidion… Atherton  &lt;NA&gt;         Falster_2005_… wood_dens… 0.566
#&gt; 5 Falster_20… Homalanthu… Atherton  &lt;NA&gt;         Falster_2005_… wood_dens… 0.319
#&gt; 6 Falster_20… Melicope e… Atherton  &lt;NA&gt;         Falster_2005_… wood_dens… 0.346
#&gt; # … with 5 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;</code></pre>
<p>Filtering <strong>all traits with leaf in the trait name</strong> and assigning it to an object</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span> <span class="va">leaf_traits</span> <span class="op">&lt;-</span> <span class="va">austraits</span><span class="op">$</span><span class="va">traits</span><span class="op">$</span><span class="va">trait_name</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">.</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"leaf"</span>,<span class="va">.</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<pre><code>#&gt; [1] "leaf_angle"          "leaf_area"           "leaf_compoundness"  
#&gt; [4] "leaf_N_per_dry_mass" "specific_leaf_area"</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_leaf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_trait.html">extract_trait</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="va">leaf_traits</span><span class="op">)</span> <span class="co"># Extract from austraits</span>

<span class="va">data_leaf</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 12
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;dbl&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  66.1
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  71.7
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; # … with 5 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;</code></pre>
</div>
</div>
<div id="join-data-from-other-tables-and-elements" class="section level2">
<h2 class="hasAnchor">
<a href="#join-data-from-other-tables-and-elements" class="anchor"></a>Join data from other tables and elements</h2>
<p>Once users have extracted the data they want, they may want to merge in other study details into the main <code>traits</code> dataframe for their analyses. For example, users may require taxonomic information for a phylogenetic analysis. This is where the <code>join_</code> functions come in.</p>
<p>There are five <code>join_</code> functions in total, each designed to append specific information from other tables and elements in the <code>austraits</code> object. Their suffixes refer to the type of information that is joined, e.g. <code>join_taxonomy</code> appends taxonomic information to the <code>traits</code> dataframe. See <code><a href="../reference/join_all.html">?join_all</a></code> for more details.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Join taxonomic information </span>
<span class="op">(</span><span class="va">data_leaf</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="va">join_taxonomy</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 16
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;dbl&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  66.1
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  71.7
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; # … with 9 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;, family &lt;chr&gt;, genus &lt;chr&gt;,
#&gt; #   taxonRank &lt;chr&gt;, acceptedNameUsageID &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Join methodological information </span>
<span class="op">(</span><span class="va">data_leaf</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="va">join_methods</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 16
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;dbl&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  66.1
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  71.7
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; # … with 9 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;, methods &lt;chr&gt;,
#&gt; #   year_collected_start &lt;chr&gt;, year_collected_end &lt;chr&gt;, collection_type &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Join site based information </span>
<span class="op">(</span><span class="va">data_leaf</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="va">join_sites</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 14
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;dbl&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  66.1
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  71.7
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; # … with 7 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;, latitude (deg) &lt;chr&gt;,
#&gt; #   longitude (deg) &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Join context information </span>
<span class="op">(</span><span class="va">data_leaf</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="va">join_contexts</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 12
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;dbl&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  66.1
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  71.7
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; # … with 5 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Alternatively users can join *all* information </span>
<span class="op">(</span><span class="va">data_leaf</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="va">join_all</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; # A tibble: 6 × 22
#&gt;   dataset_id   taxon_name site_name context_name observation_id trait_name value
#&gt;   &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;      &lt;dbl&gt;
#&gt; 1 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  66.1
#&gt; 2 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  319  
#&gt; 3 Falster_2003 Acacia my… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; 4 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_angle  71.7
#&gt; 5 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_area  562  
#&gt; 6 Falster_2003 Acacia su… Ku-ring-… &lt;NA&gt;         Falster_2003_… leaf_comp…  NA  
#&gt; # … with 15 more variables: unit &lt;chr&gt;, date &lt;chr&gt;, value_type &lt;fct&gt;,
#&gt; #   replicates &lt;chr&gt;, original_name &lt;chr&gt;, latitude (deg) &lt;chr&gt;,
#&gt; #   longitude (deg) &lt;chr&gt;, family &lt;chr&gt;, genus &lt;chr&gt;, taxonRank &lt;chr&gt;,
#&gt; #   acceptedNameUsageID &lt;chr&gt;, methods &lt;chr&gt;, year_collected_start &lt;chr&gt;,
#&gt; #   year_collected_end &lt;chr&gt;, collection_type &lt;chr&gt;</code></pre>
</div>
<div id="visualising-data-by-site" class="section level2">
<h2 class="hasAnchor">
<a href="#visualising-data-by-site" class="anchor"></a>Visualising data by site</h2>
<p><code>plot_site_locations</code> graphically summarises where trait data was collected from and how much data is available. The legend refers to the number of neighbouring points, the more warmer the colour, the more data is available. This function only works for studies that are geo-referenced. Users must first use <code>join_sites</code> to append latitude and longitude information into the trait dataframe before plotting</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_wood_dens</span> <span class="op">&lt;-</span> <span class="va">data_wood_dens</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="va">join_sites</span>
<span class="va">data_wood_dens</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_site_locations.html">plot_site_locations</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="austraits_files/figure-html/site_plot-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="visualising-data-distribution-and-variance" class="section level2">
<h2 class="hasAnchor">
<a href="#visualising-data-distribution-and-variance" class="anchor"></a>Visualising data distribution and variance</h2>
<p><code>plot_trait_distribution</code> creates histograms and <a href="https://github.com/eclarke/ggbeeswarm">beeswarm plots</a> for specific traits to help users visualise the variance of the data. Users can specific whether to create seperate beeswarm plots at the level of taxonomic family or for each <code>dataset_id</code></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">austraits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_trait_distribution_beeswarm.html">plot_trait_distribution_beeswarm</a></span><span class="op">(</span><span class="st">"wood_density"</span>, <span class="st">"family"</span><span class="op">)</span></code></pre></div>
<p><img src="austraits_files/figure-html/beeswarm-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">austraits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_trait_distribution_beeswarm.html">plot_trait_distribution_beeswarm</a></span><span class="op">(</span><span class="st">"wood_density"</span>, <span class="st">"dataset_id"</span><span class="op">)</span></code></pre></div>
<p><img src="austraits_files/figure-html/beeswarm-2.png" width="576" style="display: block; margin: auto;"></p>
<div id="pivotting-from-long-to-wide-format" class="section level3">
<h3 class="hasAnchor">
<a href="#pivotting-from-long-to-wide-format" class="anchor"></a>Pivotting from long to wide format</h3>
<p>The table of traits in AusTraits comes in long format, where data for all traits are in two columns called <code>trait_name</code> and <code>value</code>. You can convert to wide format, where each trait is in a separate column, using the function <code>trait_pivot_wider</code>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_wide</span> <span class="op">&lt;-</span> <span class="va">data_falster_studies</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="../reference/trait_pivot_wider.html">trait_pivot_wider</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>The returned object is a list with several parts:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_wide</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; [1] "value"      "unit"       "date"       "value_type" "replicates"</code></pre>
<p>Looking at the <code>value</code> element, it now has columns for each trait:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_wide</span><span class="op">$</span><span class="va">value</span> <span class="op"><a href="../reference/%&gt;%.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt;  [1] "dataset_id"           "taxon_name"           "site_name"           
#&gt;  [4] "context_name"         "observation_id"       "original_name"       
#&gt;  [7] "leaf_angle"           "leaf_area"            "leaf_compoundness"   
#&gt; [10] "branch_mass_fraction" "huber_value"          "leaf_N_per_dry_mass" 
#&gt; [13] "seed_mass"            "specific_leaf_area"   "wood_density"        
#&gt; [16] "fire_response"        "plant_height"</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_wide</span><span class="op">$</span><span class="va">value</span></code></pre></div>
<pre><code>#&gt; # A tibble: 691 × 17
#&gt;    dataset_id   taxon_name  site_name  context_name observation_id original_name
#&gt;    &lt;chr&gt;        &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;        &lt;chr&gt;          &lt;chr&gt;        
#&gt;  1 Falster_2003 Acacia myr… Ku-ring-g… &lt;NA&gt;         Falster_2003_… Acacia myrti…
#&gt;  2 Falster_2003 Acacia myr… Ku-ring-g… &lt;NA&gt;         Falster_2003_… Acacia myrti…
#&gt;  3 Falster_2003 Acacia myr… Ku-ring-g… &lt;NA&gt;         Falster_2003_… Acacia myrti…
#&gt;  4 Falster_2003 Acacia sua… Ku-ring-g… &lt;NA&gt;         Falster_2003_… Acacia suave…
#&gt;  5 Falster_2003 Acacia sua… Ku-ring-g… &lt;NA&gt;         Falster_2003_… Acacia suave…
#&gt;  6 Falster_2003 Acacia sua… Ku-ring-g… &lt;NA&gt;         Falster_2003_… Acacia suave…
#&gt;  7 Falster_2003 Angophora … Ku-ring-g… &lt;NA&gt;         Falster_2003_… Angophora hi…
#&gt;  8 Falster_2003 Angophora … Ku-ring-g… &lt;NA&gt;         Falster_2003_… Angophora hi…
#&gt;  9 Falster_2003 Angophora … Ku-ring-g… &lt;NA&gt;         Falster_2003_… Angophora hi…
#&gt; 10 Falster_2003 Banksia ma… Ku-ring-g… &lt;NA&gt;         Falster_2003_… Banksia marg…
#&gt; # … with 681 more rows, and 11 more variables: leaf_angle &lt;chr&gt;,
#&gt; #   leaf_area &lt;chr&gt;, leaf_compoundness &lt;chr&gt;, branch_mass_fraction &lt;chr&gt;,
#&gt; #   huber_value &lt;chr&gt;, leaf_N_per_dry_mass &lt;chr&gt;, seed_mass &lt;chr&gt;,
#&gt; #   specific_leaf_area &lt;chr&gt;, wood_density &lt;chr&gt;, fire_response &lt;chr&gt;,
#&gt; #   plant_height &lt;chr&gt;</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Falster, Fonti Kar, Dony Indiarto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
