<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>austraits • austraits</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="austraits">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">austraits</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">3.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/austraits.html">Get Started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/traitecoevo/austraits/"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>austraits</h1>
                        <h4 data-toc-skip class="author">Fonti Kar,
Elizabeth Wenk, Daniel Falster</h4>
            
            <h4 data-toc-skip class="date">2025-03-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/traitecoevo/austraits/blob/master/vignettes/austraits.Rmd" class="external-link"><code>vignettes/austraits.Rmd</code></a></small>
      <div class="d-none name"><code>austraits.Rmd</code></div>
    </div>

    
    
<p><code>austraits</code> allow users to <strong>access, explore and
wrangle data</strong> from <a href="https://github.com/traitecoevo/traits.build" class="external-link">traits.build</a>
relational databases. It is also an R interface to <a href="https://austraits.org/" class="external-link">AusTraits</a>, the Australian plant trait
database. This package contains functions for joining data from various
tables, filtering to specific records, combining multiple databases and
visualising the distribution of the data. Below, we’ve include a
tutorial using the AusTraits database to illustrate how some these
functions work together to generate useful outputs.</p>
<div class="section level2">
<h2 id="install-and-load-austraits">Install and load <code>austraits</code><a class="anchor" aria-label="anchor" href="#install-and-load-austraits"></a>
</h2>
<p><code>austraits</code> is still under development. To install the
current version from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"traitecoevo/austraits"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, upgrade <span class="op">=</span> <span class="st">"ask"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the austraits package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://traitecoevo.github.io/austraits/">austraits</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="retrieve-austraits-database">Retrieve AusTraits database<a class="anchor" aria-label="anchor" href="#retrieve-austraits-database"></a>
</h3>
<p>We will use the latest AusTraits database as an example database.</p>
<p>We can download the AusTraits database by calling
<code><a href="../reference/load_austraits.html">load_austraits()</a></code>. This function will download AusTraits to
a specified path. By default it is <code>data/austraits</code>. The
function will reload the database from this location in the future. You
can set <code>update = TRUE</code> so the database is downloaded fresh
from <a href="https://zenodo.org/record/3568429" class="external-link">Zenodo</a>. Note that
<code><a href="../reference/load_austraits.html">load_austraits()</a></code> will happily accept a DOI of a particular
version.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">austraits</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_austraits.html">load_austraits</a></span><span class="op">(</span>version <span class="op">=</span> <span class="st">"6.0.0"</span>, path <span class="op">=</span> <span class="st">"data/austraits"</span><span class="op">)</span></span></code></pre></div>
<p>You can check out different versions of AusTraits and their
associated DOI by using:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_versions.html">get_versions</a></span><span class="op">(</span>path <span class="op">=</span> <span class="st">"data/austraits"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 6 × 4</span></span>
<span><span class="co">#&gt;   publication_date doi                     version id      </span></span>
<span><span class="co">#&gt;   &lt;date&gt;           &lt;chr&gt;                   &lt;chr&gt;   &lt;chr&gt;   </span></span>
<span><span class="co">#&gt; 1 2024-05-14       10.5281/zenodo.11188867 6.0.0   11188867</span></span>
<span><span class="co">#&gt; 2 2023-11-19       10.5281/zenodo.10156222 5.0.0   10156222</span></span>
<span><span class="co">#&gt; 3 2023-09-18       10.5281/zenodo.8353840  4.2.0   8353840 </span></span>
<span><span class="co">#&gt; 4 2023-01-30       10.5281/zenodo.7583087  4.1.0   7583087 </span></span>
<span><span class="co">#&gt; 5 2022-11-27       10.5281/zenodo.7368074  4.0.0   7368074 </span></span>
<span><span class="co">#&gt; 6 2021-07-14       10.5281/zenodo.5112001  3.0.2   5112001</span></span></code></pre>
<p>AusTraits, like all traits.build databases, is a relational database.
In R, it is a very <em>large</em> list with multiple tables. If you are
not familiar with working with lists in R, we recommend having a quick
look at this <a href="https://www.tutorialspoint.com/r/r_lists.htm" class="external-link">tutorial</a>. To
learn more about the structure of <code>austraits</code>, check out the
<a href="https://traitecoevo.github.io/austraits/articles/structure.html">structure
of the database</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">austraits</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; ── This is 6.0.0 of AusTraits: a curated plant trait database for the Australian flora! ──────────────────────────────</span></span>
<span></span>
<span><span class="co">#&gt; ℹ This database is built using traits.build version 1.1.0.9000</span></span>
<span><span class="co">#&gt; ℹ This database contains a total of 1726024 records, for 33494 taxa and 497 traits.</span></span>
<span></span>
<span><span class="co">#&gt; ── This object is a 'list' with the following components: ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; • traits: A table containing measurements of traits.</span></span>
<span><span class="co">#&gt; • locations: A table containing observations of location/site characteristics associated with information in</span></span>
<span><span class="co">#&gt; `traits`. Cross referencing between the two dataframes is possible using combinations of the variables `dataset_id`,</span></span>
<span><span class="co">#&gt; `location_name`.</span></span>
<span><span class="co">#&gt; • contexts: A table containing observations of contextual characteristics associated with information in `traits`.</span></span>
<span><span class="co">#&gt; Cross referencing between the two dataframes is possible using combinations of the variables `dataset_id`, `link_id`,</span></span>
<span><span class="co">#&gt; and `link_vals`.</span></span>
<span><span class="co">#&gt; • methods: A table containing details on methods with which data were collected, including time frame and source.</span></span>
<span><span class="co">#&gt; Cross referencing with the `traits` table is possible using combinations of the variables `dataset_id`, `trait_name`.</span></span>
<span><span class="co">#&gt; • excluded_data: A table of data that did not pass quality test and so were excluded from the master dataset.</span></span>
<span><span class="co">#&gt; • taxonomic_updates: A table of all taxonomic changes implemented in the construction of AusTraits. Changes are</span></span>
<span><span class="co">#&gt; determined by comparing against the APC (Australian Plant Census) and APNI (Australian Plant Names Index).</span></span>
<span><span class="co">#&gt; • taxa: A table containing details on taxa associated with information in `traits`. This information has been sourced</span></span>
<span><span class="co">#&gt; from the APC (Australian Plant Census) and APNI (Australian Plant Names Index) and is released under a CC-BY3</span></span>
<span><span class="co">#&gt; license.</span></span>
<span><span class="co">#&gt; • contributors: A table of people contributing to each study.</span></span>
<span><span class="co">#&gt; • sources: Bibtex entries for all primary and secondary sources in the compilation.</span></span>
<span><span class="co">#&gt; • definitions: A copy of the definitions for all tables and terms. Information included here was used to process data</span></span>
<span><span class="co">#&gt; and generate any documentation for the study.</span></span>
<span><span class="co">#&gt; • schema: A copy of the schema for all tables and terms. Information included here was used to process data and</span></span>
<span><span class="co">#&gt; generate any documentation for the study.</span></span>
<span><span class="co">#&gt; • metadata: Metadata associated with the dataset, including title, creators, license, subject, funding sources.</span></span>
<span><span class="co">#&gt; • build_info: A description of the computing environment used to create this version of the dataset, including</span></span>
<span><span class="co">#&gt; version number, git commit and R session_info.</span></span>
<span></span>
<span><span class="co">#&gt; ℹ To access a component, try using the $ e.g. austraits$traits</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="descriptive-summaries-of-traits-and-taxa">Descriptive summaries of traits and taxa<a class="anchor" aria-label="anchor" href="#descriptive-summaries-of-traits-and-taxa"></a>
</h2>
<p>AusTraits contains 497 plant traits. Check out <a href="https://w3id.org/APD" class="external-link">definitions of the traits</a> to learn more
about how each trait is defined.</p>
<p>Have a look at data coverage by trait or taxa with:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/summarise_database.html">summarise_database</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"trait_name"</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 497 × 5</span></span>
<span><span class="co">#&gt;    trait_name                    n_records n_dataset n_taxa percent_total</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                             &lt;int&gt;     &lt;int&gt;  &lt;int&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 accessory_cost_fraction              47         1     47     0.0000272</span></span>
<span><span class="co">#&gt;  2 accessory_cost_mass                  47         1     47     0.0000272</span></span>
<span><span class="co">#&gt;  3 atmospheric_CO2_concentration       840         4    121     0.000487 </span></span>
<span><span class="co">#&gt;  4 bark_Al_per_dry_mass                 70         1     10     0.0000406</span></span>
<span><span class="co">#&gt;  5 bark_B_per_dry_mass                  70         1     10     0.0000406</span></span>
<span><span class="co">#&gt;  6 bark_C_per_dry_mass                 229         2     27     0.000133 </span></span>
<span><span class="co">#&gt;  7 bark_Ca_per_dry_mass                104         3     21     0.0000603</span></span>
<span><span class="co">#&gt;  8 bark_Cu_per_dry_mass                 70         1     10     0.0000406</span></span>
<span><span class="co">#&gt;  9 bark_Fe_per_dry_mass                 70         1     10     0.0000406</span></span>
<span><span class="co">#&gt; 10 bark_K_per_dry_mass                 104         3     21     0.0000603</span></span>
<span><span class="co">#&gt; # ℹ 487 more rows</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/summarise_database.html">summarise_database</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"family"</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 310 × 5</span></span>
<span><span class="co">#&gt;    family           n_records n_dataset n_taxa percent_total</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                &lt;int&gt;     &lt;int&gt;  &lt;int&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 Acanthaceae           3719        57    149     0.00216  </span></span>
<span><span class="co">#&gt;  2 Achariaceae            162        14      3     0.0000939</span></span>
<span><span class="co">#&gt;  3 Actinidiaceae          186        16      3     0.000108 </span></span>
<span><span class="co">#&gt;  4 Agapanthaceae          107        13      3     0.000062 </span></span>
<span><span class="co">#&gt;  5 Aizoaceae             5004        63    102     0.0029   </span></span>
<span><span class="co">#&gt;  6 Akaniaceae             123        16      1     0.0000713</span></span>
<span><span class="co">#&gt;  7 Alismataceae           892        30     20     0.000517 </span></span>
<span><span class="co">#&gt;  8 Alliaceae              561        19     18     0.000325 </span></span>
<span><span class="co">#&gt;  9 Alseuosmiaceae         318        13      3     0.000184 </span></span>
<span><span class="co">#&gt; 10 Alstroemeriaceae       175        15      2     0.000101 </span></span>
<span><span class="co">#&gt; # ℹ 300 more rows</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/summarise_database.html">summarise_database</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"genus"</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 3,177 × 5</span></span>
<span><span class="co">#&gt;    genus        n_records n_dataset n_taxa percent_total</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;            &lt;int&gt;     &lt;int&gt;  &lt;int&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 (Dockrillia          3         2      1    0.00000174</span></span>
<span><span class="co">#&gt;  2 Abelia              16         4      1    0.00000928</span></span>
<span><span class="co">#&gt;  3 Abelmoschus        271        19      8    0.000157  </span></span>
<span><span class="co">#&gt;  4 Abildgaardia        74         7      2    0.0000429 </span></span>
<span><span class="co">#&gt;  5 Abrodictyum        123        14      3    0.0000713 </span></span>
<span><span class="co">#&gt;  6 Abroma              39         7      2    0.0000226 </span></span>
<span><span class="co">#&gt;  7 Abrophyllum        181        19      3    0.000105  </span></span>
<span><span class="co">#&gt;  8 Abrotanella        183        18      4    0.000106  </span></span>
<span><span class="co">#&gt;  9 Abrus              202        26      3    0.000117  </span></span>
<span><span class="co">#&gt; 10 Abutilon          1975        52     54    0.00115   </span></span>
<span><span class="co">#&gt; # ℹ 3,167 more rows</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="quickly-look-up-data">Quickly look up data<a class="anchor" aria-label="anchor" href="#quickly-look-up-data"></a>
</h2>
<p>Interested in a specific trait or context property, but unsure what
terms we use? Try our <code>lookup_</code> functions.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/lookup_trait.html">lookup_trait</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"leaf"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "leaf_compoundness" "leaf_phenology"    "leaf_length"       "leaf_width"        "leaf_margin"      </span></span>
<span><span class="co">#&gt; [6] "leaf_shape"</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/lookup_context_property.html">lookup_context_property</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"fire"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "fire intensity"     "fire history"       "fire response type" "fire severity"      "fire season"</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/lookup_location_property.html">lookup_location_property</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"temperature"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "temperature, max (C)"             "temperature, MAT (C)"             "temperature, mean summer max (C)"</span></span>
<span><span class="co">#&gt; [4] "temperature, mean winter max (C)" "temperature, max MAT (C)"         "temperature, min MAT (C)"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="extracting-data">Extracting data<a class="anchor" aria-label="anchor" href="#extracting-data"></a>
</h2>
<p>In most cases, users would like to extract a subset of a database for
their research purposes.</p>
<ul>
<li>
<code><a href="../reference/extract_dataset.html">extract_dataset()</a></code> filters for a particular study</li>
<li>
<code><a href="../reference/extract_trait.html">extract_trait()</a></code> filters for a certain trait</li>
<li>
<code><a href="../reference/extract_taxa.html">extract_taxa()</a></code> filters for a specific taxon</li>
</ul>
<p>Note you can supply a vector to each of these functions to filter for
more than one study/trait/taxa. All our <code>extract_</code> function
supports partial matching e.g. <code>extract_trait("leaf")</code> would
return all traits containing <code>leaf</code>.</p>
<p>If you would like to extract from other tables or columns, use <a href="#extract-data"><code>extract_data</code></a></p>
<p>All <code>extract_</code> functions simultaneously filter across all
tables in the database.</p>
<div class="section level3">
<h3 id="extracting-by-dataset">Extracting by dataset<a class="anchor" aria-label="anchor" href="#extracting-by-dataset"></a>
</h3>
<p>Filtering <strong>one particular dataset</strong> and assigning it to
an object</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">one_study</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_dataset.html">extract_dataset</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"Falster_2005_2"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">one_study</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 165 × 26</span></span>
<span><span class="co">#&gt;    dataset_id     taxon_name    observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;          &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Falster_2005_2 Acacia longi… 01             huber_val… 0.00… mm2{… population  mean       measurement    unknown   </span></span>
<span><span class="co">#&gt;  2 Falster_2005_2 Acacia longi… 01             huber_val… 0.00… mm2{… population  mean       measurement    unknown   </span></span>
<span><span class="co">#&gt;  3 Falster_2005_2 Acacia longi… 01             huber_val… 0.00… mm2{… population  mean       measurement    unknown   </span></span>
<span><span class="co">#&gt;  4 Falster_2005_2 Acacia longi… 01             huber_val… 0.00… mm2{… population  mean       measurement    unknown   </span></span>
<span><span class="co">#&gt;  5 Falster_2005_2 Acacia longi… 01             leaf_N_pe… 23.2  mg/g  population  mean       measurement    4         </span></span>
<span><span class="co">#&gt;  6 Falster_2005_2 Acacia longi… 01             leaf_area  1761  mm2   population  mean       measurement    4         </span></span>
<span><span class="co">#&gt;  7 Falster_2005_2 Acacia longi… 01             leaf_mass… 128   g/m2  population  mean       measurement    4         </span></span>
<span><span class="co">#&gt;  8 Falster_2005_2 Acacia longi… 01             plant_hei… 4     m     population  maximum    measurement    unknown   </span></span>
<span><span class="co">#&gt;  9 Falster_2005_2 Acacia longi… 01             resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Falster_2005_2 Acacia longi… 01             seed_dry_… 14    mg    population  mean       measurement    unknown   </span></span>
<span><span class="co">#&gt; # ℹ 155 more rows</span></span>
<span><span class="co">#&gt; # ℹ 16 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;</span></span></code></pre>
<p>Filtering <strong>multiple datasets</strong> and assigning it to an
object</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multi_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_dataset.html">extract_dataset</a></span><span class="op">(</span><span class="va">austraits</span>, </span>
<span>                                        dataset_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Thompson_2001"</span>,<span class="st">"Ilic_2000"</span><span class="op">)</span><span class="op">)</span></span>
<span> </span>
<span><span class="va">multi_studies</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 2,209 × 26</span></span>
<span><span class="co">#&gt;    dataset_id taxon_name       observation_id trait_name  value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;            &lt;chr&gt;          &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Ilic_2000  Acacia acradenia 0001           wood_densi… 0.904 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  2 Ilic_2000  Acacia acuminata 0002           wood_densi… 0.895 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  3 Ilic_2000  Acacia acuminata 0003           wood_densi… 1.008 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  4 Ilic_2000  Acacia adsurgens 0004           wood_densi… 0.887 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  5 Ilic_2000  Acacia alleniana 0005           wood_densi… 0.56  mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  6 Ilic_2000  Acacia ampliceps 0006           wood_densi… 0.568 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  7 Ilic_2000  Acacia aneura    0007           wood_densi… 1.035 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  8 Ilic_2000  Acacia aneura    0008           wood_densi… 1.019 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt;  9 Ilic_2000  Acacia aneura    0009           wood_densi… 0.861 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt; 10 Ilic_2000  Acacia aneura    0010           wood_densi… 0.996 mg/m… individual  raw        measurement    unknown   </span></span>
<span><span class="co">#&gt; # ℹ 2,199 more rows</span></span>
<span><span class="co">#&gt; # ℹ 16 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;</span></span></code></pre>
<p>Filtering <strong>multiple datasets by same lead author</strong>
(e.g. Falster) and assigning it to an object.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">falster_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_dataset.html">extract_dataset</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"Falster"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">falster_studies</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 685 × 26</span></span>
<span><span class="co">#&gt;    dataset_id   taxon_name      observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;        &lt;chr&gt;           &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Falster_2003 Acacia floribu… 01             leaf_area  142   mm2   population  mean       measurement    3         </span></span>
<span><span class="co">#&gt;  2 Falster_2003 Acacia floribu… 01             leaf_incl… 57    deg   population  mean       measurement    3         </span></span>
<span><span class="co">#&gt;  3 Falster_2003 Acacia floribu… 02             leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Falster_2003 Acacia myrtifo… 03             leaf_area  319   mm2   population  mean       measurement    3         </span></span>
<span><span class="co">#&gt;  5 Falster_2003 Acacia myrtifo… 03             leaf_incl… 66.1  deg   population  mean       measurement    3         </span></span>
<span><span class="co">#&gt;  6 Falster_2003 Acacia myrtifo… 04             leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Falster_2003 Acacia suaveol… 05             leaf_area  562   mm2   population  mean       measurement    3         </span></span>
<span><span class="co">#&gt;  8 Falster_2003 Acacia suaveol… 05             leaf_incl… 71.7  deg   population  mean       measurement    3         </span></span>
<span><span class="co">#&gt;  9 Falster_2003 Acacia suaveol… 06             leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Falster_2003 Angophora hisp… 07             leaf_area  1590  mm2   population  mean       measurement    3         </span></span>
<span><span class="co">#&gt; # ℹ 675 more rows</span></span>
<span><span class="co">#&gt; # ℹ 16 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="extracting-by-taxonomy">Extracting by taxonomy<a class="anchor" aria-label="anchor" href="#extracting-by-taxonomy"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># By family </span></span>
<span><span class="va">proteaceae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_taxa.html">extract_taxa</a></span><span class="op">(</span><span class="va">austraits</span>, family <span class="op">=</span> <span class="st">"Proteaceae"</span><span class="op">)</span></span>
<span><span class="co"># Checking that only taxa in Proteaceae have been extracted</span></span>
<span><span class="va">proteaceae</span><span class="op">$</span><span class="va">taxa</span><span class="op">$</span><span class="va">family</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "Proteaceae"</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># By genus </span></span>
<span><span class="va">acacia</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_taxa.html">extract_taxa</a></span><span class="op">(</span><span class="va">austraits</span>, genus <span class="op">=</span> <span class="st">"Acacia"</span><span class="op">)</span></span>
<span><span class="co"># Checking that only taxa in Acacia have been extracted</span></span>
<span><span class="va">acacia</span><span class="op">$</span><span class="va">traits</span><span class="op">$</span><span class="va">taxon_name</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "Acacia abbatiana"                        "Acacia abbreviata"                      </span></span>
<span><span class="co">#&gt; [3] "Acacia abrupta"                          "Acacia acanthaster"                     </span></span>
<span><span class="co">#&gt; [5] "Acacia acanthoclada subsp. acanthoclada" "Acacia acanthoclada subsp. glaucescens"</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acacia</span><span class="op">$</span><span class="va">taxa</span><span class="op">$</span><span class="va">genus</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "Acacia"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="extracting-by-trait">Extracting by trait<a class="anchor" aria-label="anchor" href="#extracting-by-trait"></a>
</h3>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_fruit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_trait.html">extract_trait</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"fruit"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_fruit</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 216,465 × 26</span></span>
<span><span class="co">#&gt;    dataset_id taxon_name        observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;             &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 ABRS_1981  Ceratophyllum de… 0566           fruit_len… 4     mm    species     minimum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 ABRS_1981  Ceratophyllum de… 0566           fruit_len… 6     mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 ABRS_1981  Ceratophyllum de… 0566           fruit_wid… 3     mm    species     minimum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 ABRS_1981  Ceratophyllum de… 0566           fruit_wid… 3.5   mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 ABRS_1981  Conospermum peti… 0680           fruit_len… 2.5   mm    species     minimum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 ABRS_1981  Conospermum peti… 0680           fruit_wid… 3     mm    species     minimum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 ABRS_1981  Proiphys amboine… 3182           fruit_len… 15    mm    species     minimum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 ABRS_1981  Proiphys amboine… 3182           fruit_len… 30    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 ABRS_1981  Proiphys amboine… 3182           fruit_wid… 15    mm    species     minimum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 ABRS_1981  Proiphys amboine… 3182           fruit_wid… 30    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 216,455 more rows</span></span>
<span><span class="co">#&gt; # ℹ 16 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;</span></span></code></pre>
<p>Combining <code>lookup_trait</code> with <code>extract_trait</code>
to obtain <strong>all traits with ‘leaf’ in the trait name</strong> and
assigning it to an object. Note we use the <code>.</code> notation to
pass on the <code>lookup_trait</code> results to
<code>extract_trait</code></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">leaf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lookup_trait.html">lookup_trait</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="st">"leaf"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/extract_trait.html">extract_trait</a></span><span class="op">(</span><span class="va">austraits</span>, <span class="va">.</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">leaf</span><span class="op">$</span><span class="va">traits</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 511,952 × 26</span></span>
<span><span class="co">#&gt;    dataset_id taxon_name        observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;             &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 ABRS_1981  Acanthocarpus ca… 0001           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 ABRS_1981  Acanthocarpus hu… 0002           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 ABRS_1981  Acanthocarpus pa… 0003           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 ABRS_1981  Acanthocarpus pr… 0004           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 ABRS_1981  Acanthocarpus ro… 0005           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 ABRS_1981  Acanthocarpus ru… 0006           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 ABRS_1981  Acanthocarpus ve… 0007           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 ABRS_1981  Acer pseudoplata… 0008           leaf_phen… deci… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 ABRS_1981  Acidonia microca… 0009           leaf_comp… comp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 ABRS_1981  Callitris acumin… 0010           leaf_comp… simp… &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 511,942 more rows</span></span>
<span><span class="co">#&gt; # ℹ 16 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="extract-data">Extracting from other tables<a class="anchor" aria-label="anchor" href="#extract-data"></a>
</h3>
<p>You may want to extract data from tables that have specific column
values. For example calling the code below will return data where “fire”
is mentioned in the <code>context_property</code> column</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_fire</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span><span class="va">austraits</span>, </span>
<span>                          table <span class="op">=</span>  <span class="st">"contexts"</span>,</span>
<span>                          col <span class="op">=</span>  <span class="st">"context_property"</span>, </span>
<span>                          col_value <span class="op">=</span> <span class="st">"fire"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_fire</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extracting-from-a-single-table">Extracting from a single table<a class="anchor" aria-label="anchor" href="#extracting-from-a-single-table"></a>
</h3>
<p>If you have already manipulated the original database and are working
with just the traits table, the extract functions will also work on a
single table.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seedling_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span><span class="va">austraits</span><span class="op">$</span><span class="va">traits</span>,</span>
<span>                          col <span class="op">=</span>  <span class="st">"life_stage"</span>, </span>
<span>                          col_value <span class="op">=</span> <span class="st">"seedling"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Falster_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span><span class="va">austraits</span><span class="op">$</span><span class="va">traits</span>,</span>
<span>                          col <span class="op">=</span>  <span class="st">"dataset_id"</span>, </span>
<span>                          col_value <span class="op">=</span> <span class="st">"Falster"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">leaf_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_trait.html">extract_trait</a></span><span class="op">(</span><span class="va">austraits</span><span class="op">$</span><span class="va">traits</span>, </span>
<span>                          <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"leaf_area"</span>, <span class="st">"leaf_N_per_dry_mass"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="join-data-from-other-tables">Join data from other tables<a class="anchor" aria-label="anchor" href="#join-data-from-other-tables"></a>
</h2>
<p>Once users have extracted the data they want, they may want to merge
other study details into the main <code>traits</code> dataframe for
their analyses. For example, users may require taxonomic information for
a phylogenetic analysis. This is where the <code>join_</code> functions
come in.</p>
<p>There are five <code>join_</code> functions in total, each designed
to append specific information from other tables and elements in the
<code>austraits</code> object. Their suffixes refer to the type of
information that is joined, e.g. <code>join_taxa</code> appends
taxonomic information to the <code>traits</code> dataframe.</p>
<ul>
<li><code><a href="../reference/join_taxa.html">join_taxa()</a></code></li>
<li><code><a href="../reference/join_methods.html">join_methods()</a></code></li>
<li><code><a href="../reference/join_location_coordinates.html">join_location_coordinates()</a></code></li>
<li><code><a href="../reference/join_location_properties.html">join_location_properties()</a></code></li>
<li><code><a href="../reference/join_context_properties.html">join_context_properties()</a></code></li>
</ul>
<p>We recommend pulling up the help file for each one for more details
e.g <code>?join_location_coordinates()</code></p>
<p>Each of the functions has specific default parameters and formatting,
but offers versatile joining options.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join taxonomic information </span></span>
<span><span class="op">(</span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">join_taxa</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 30</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 20 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;, family &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   genus &lt;chr&gt;, taxon_rank &lt;chr&gt;, establishment_means &lt;chr&gt;</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join methodological information </span></span>
<span><span class="op">(</span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">join_methods</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 27</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 17 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;, methods &lt;chr&gt;</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join location coordinates </span></span>
<span><span class="op">(</span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">join_location_coordinates</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 29</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 19 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;, location_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   `latitude (deg)` &lt;chr&gt;, `longitude (deg)` &lt;chr&gt;</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join information pertaining to location properties </span></span>
<span><span class="op">(</span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">join_location_properties</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 28</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 18 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;, location_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   location_properties &lt;chr&gt;</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join information pertaining to location properties </span></span>
<span><span class="op">(</span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/join_location_properties.html">join_location_properties</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"many_columns"</span>, vars <span class="op">=</span> <span class="st">"temperature, min MAT (C)"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 28</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 18 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;, location_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   `location_property: temperature, min MAT (C)` &lt;chr&gt;</span></span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join context information </span></span>
<span><span class="op">(</span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">join_context_properties</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 31</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 21 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   treatment_context_properties &lt;chr&gt;, plot_context_properties &lt;chr&gt;, entity_context_properties &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   temporal_context_properties &lt;chr&gt;, method_context_properties &lt;chr&gt;</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join information from multiple tables </span></span>
<span><span class="op">(</span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">join_context_properties</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">join_location_coordinates</span><span class="op">)</span><span class="op">$</span><span class="va">traits</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 34</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 24 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   treatment_context_properties &lt;chr&gt;, plot_context_properties &lt;chr&gt;, entity_context_properties &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   temporal_context_properties &lt;chr&gt;, method_context_properties &lt;chr&gt;, location_name &lt;chr&gt;, …</span></span></code></pre>
<p>Alternatively,users can join <strong>all</strong> information using
<code><a href="../reference/flatten_database.html">flatten_database()</a></code>:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/flatten_database.html">flatten_database</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,822 × 66</span></span>
<span><span class="co">#&gt;    dataset_id    taxon_name     observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;          &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Campbell_2006 Acacia falcif… 001            bud_bank_… basa… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 Campbell_2006 Acacia falcif… 001            resprouti… resp… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 Campbell_2006 Acacia falcif… 001            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 Campbell_2006 Acacia falcif… 002            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 Campbell_2006 Acacia falcif… 003            dispersers ants  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 Campbell_2006 Acacia falcif… 003            plant_gro… tree  &lt;NA&gt;  species     mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 Campbell_2006 Acacia irrora… 004            bud_bank_… none  &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 Campbell_2006 Acacia irrora… 004            resprouti… fire… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 Campbell_2006 Acacia irrora… 004            seedbank_… soil… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 Campbell_2006 Acacia irrora… 005            post_fire… post… &lt;NA&gt;  population  mode       expert_score   &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 1,812 more rows</span></span>
<span><span class="co">#&gt; # ℹ 56 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;, location_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   `latitude (deg)` &lt;chr&gt;, `longitude (deg)` &lt;chr&gt;, location_properties &lt;chr&gt;, treatment_context_properties &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   plot_context_properties &lt;chr&gt;, entity_context_properties &lt;chr&gt;, temporal_context_properties &lt;chr&gt;, …</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="visualising-data-by-site">Visualising data by site<a class="anchor" aria-label="anchor" href="#visualising-data-by-site"></a>
</h2>
<p><code><a href="../reference/plot_locations.html">plot_locations()</a></code> graphically summarises where trait data
was collected from and how much data is available. The legend refers to
the number of neighbouring points: the warmer the colour, the more data
that is available. This function only works for studies that are
geo-referenced. Users must first use
<code><a href="../reference/join_location_coordinates.html">join_location_coordinates()</a></code> to append latitude and
longitude information from the locations dataframe into the traits
dataframe before plotting.</p>
<p><code><a href="../reference/plot_locations.html">plot_locations()</a></code> defaults to dividing the data by
trait_name (feature = “trait_name”), but you can select any of the
columns within the traits table - including columns you add with
<code>join_</code> functions. However, selecting <code>taxon_name</code>
will likely crash R if you are working with a dataframe that still
contains a large number of species.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_fire</span> <span class="op">&lt;-</span> <span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/join_location_coordinates.html">join_location_coordinates</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_locations.html">plot_locations</a></span><span class="op">(</span><span class="va">data_fire</span><span class="op">$</span><span class="va">traits</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="reference/figures/site_plot-1.png" alt="plot of chunk site_plot"><p class="caption">
plot of chunk site_plot
</p>
</div>
</div>
<div class="section level2">
<h2 id="visualising-data-distribution-and-variance">Visualising data distribution and variance<a class="anchor" aria-label="anchor" href="#visualising-data-distribution-and-variance"></a>
</h2>
<p><code>plot_trait_distribution()</code> creates histograms and <a href="https://github.com/eclarke/ggbeeswarm" class="external-link">beeswarm plots</a> for
specific traits to help users visualise the variance of the data. Users
can specify whether to create separate beeswarm plots at the level of
taxonomic family, genus or by a column in the traits table, such as
<code>dataset_id</code></p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">austraits</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_trait_distribution_beeswarm.html">plot_trait_distribution_beeswarm</a></span><span class="op">(</span>trait_name <span class="op">=</span> <span class="st">"wood_density"</span>, y_axis_category <span class="op">=</span> <span class="st">"family"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="reference/figures/beeswarm-1.png" alt="plot of chunk beeswarm"><p class="caption">
plot of chunk beeswarm
</p>
</div>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">austraits</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_trait_distribution_beeswarm.html">plot_trait_distribution_beeswarm</a></span><span class="op">(</span>trait_name <span class="op">=</span> <span class="st">"wood_density"</span>, y_axis_category <span class="op">=</span> <span class="st">"dataset_id"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="reference/figures/beeswarm-2.png" alt="plot of chunk beeswarm"><p class="caption">
plot of chunk beeswarm
</p>
</div>
</div>
<div class="section level2">
<h2 id="reshaping-the-traits-table">Reshaping the traits table<a class="anchor" aria-label="anchor" href="#reshaping-the-traits-table"></a>
</h2>
<p>The traits table in AusTraits is in <strong>long</strong> format,
where data for all trait information are denoted by two columns called
<code>trait_name</code> and <code>value</code>. You can convert this to
wide format, where each trait is in a separate column, using the
function <code><a href="../reference/trait_pivot_wider.html">trait_pivot_wider()</a></code>.</p>
<p>Note that the following columns are lost when pivoting: unit,
replicates, measurement_remarks, and basis_of_value to provide a useful
output.</p>
<div class="section level3">
<h3 id="pivot-wider">Pivot wider<a class="anchor" aria-label="anchor" href="#pivot-wider"></a>
</h3>
<p><strong>Note</strong> that the latest version of
<code><a href="../reference/trait_pivot_wider.html">trait_pivot_wider()</a></code> is no longer supporting AusTraits
database versions &lt;=4.0.2. Please refer to our <a href="https://github.com/traitecoevo/austraits" class="external-link">README</a> to install an
older version of the <code>austraits</code> R package to work old
versions of the AusTraits database.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_fire</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/trait_pivot_wider.html">trait_pivot_wider</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 1,366 × 49</span></span>
<span><span class="co">#&gt;    dataset_id  taxon_name observation_id entity_type value_type basis_of_record life_stage population_id individual_id</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;      &lt;chr&gt;         &lt;chr&gt;        </span></span>
<span><span class="co">#&gt;  1 Campbell_2… Acacia fa… 001            population  mode       field           adult      01            &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  2 Campbell_2… Acacia fa… 002            population  mode       field           seedling   01            &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  3 Campbell_2… Acacia fa… 003            species     mode       field           adult      &lt;NA&gt;          &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  4 Campbell_2… Acacia ir… 004            population  mode       field           adult      01            &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  5 Campbell_2… Acacia ir… 005            population  mode       field           seedling   01            &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  6 Campbell_2… Acacia ir… 006            species     mode       field           adult      &lt;NA&gt;          &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  7 Campbell_2… Acacia ma… 007            population  mode       field           adult      02            &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  8 Campbell_2… Acacia ma… 008            population  mode       field           seedling   02            &lt;NA&gt;         </span></span>
<span><span class="co">#&gt;  9 Campbell_2… Acacia ma… 009            species     mode       field           adult      &lt;NA&gt;          &lt;NA&gt;         </span></span>
<span><span class="co">#&gt; 10 Campbell_2… Acacia me… 010            population  mode       field           adult      02            &lt;NA&gt;         </span></span>
<span><span class="co">#&gt; # ℹ 1,356 more rows</span></span>
<span><span class="co">#&gt; # ℹ 40 more variables: repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;, location_name &lt;chr&gt;, `latitude (deg)` &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   `longitude (deg)` &lt;chr&gt;, bud_bank_location &lt;chr&gt;, resprouting_capacity &lt;chr&gt;, seedbank_location &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   post_fire_recruitment &lt;chr&gt;, dispersers &lt;chr&gt;, plant_growth_form &lt;chr&gt;, stem_dark_respiration_per_area &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   bark_thickness &lt;chr&gt;, huber_value &lt;chr&gt;, leaf_dry_matter_content &lt;chr&gt;, leaf_dark_respiration_per_area &lt;chr&gt;, …</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="binding-trait-values">Binding trait values<a class="anchor" aria-label="anchor" href="#binding-trait-values"></a>
</h3>
<p>Some datasets will have <em>multiple observations</em> for some
traits, for instance datasets from floras often report a minimum and
maximum fruit length for a species. You can use
<code>bind_trait_values</code> to merge these into a single cell.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_fruit</span> <span class="op">&lt;-</span> <span class="va">austraits</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/extract_trait.html">extract_trait</a></span><span class="op">(</span><span class="st">"fruit_length"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/extract_taxa.html">extract_taxa</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"Rutaceae"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span>table <span class="op">=</span> <span class="st">"traits"</span>, col <span class="op">=</span> <span class="st">"value_type"</span>, col_value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"minimum"</span>, <span class="st">"maximum"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_trait_bound</span> <span class="op">&lt;-</span> <span class="va">data_fruit</span><span class="op">$</span><span class="va">traits</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/bind_trait_values.html">bind_trait_values</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Joining multiple obs with `--`</span></span>
<span>  </span>
<span><span class="va">data_trait_bound</span>  <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>   </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">value</span>, <span class="st">"--"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 288 × 26</span></span>
<span><span class="co">#&gt;    dataset_id taxon_name        observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;             &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 ABRS_2023  Acronychia aberr… 01324          fruit_len… 13--… mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  2 ABRS_2023  Acronychia acidu… 01325          fruit_len… 13--… mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  3 ABRS_2023  Acronychia acron… 01326          fruit_len… 8--13 mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  4 ABRS_2023  Acronychia acumi… 01327          fruit_len… 12--… mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  5 ABRS_2023  Acronychia baeue… 01328          fruit_len… 10--… mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  6 ABRS_2023  Acronychia choor… 01329          fruit_len… 10--… mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  7 ABRS_2023  Acronychia crass… 01330          fruit_len… 10--… mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  8 ABRS_2023  Acronychia imper… 01332          fruit_len… 9--16 mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt;  9 ABRS_2023  Acronychia laevis 01333          fruit_len… 7--10 mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt; 10 ABRS_2023  Acronychia litto… 01334          fruit_len… 8--14 mm    species     minimum--… measurement--… NA--NA    </span></span>
<span><span class="co">#&gt; # ℹ 278 more rows</span></span>
<span><span class="co">#&gt; # ℹ 16 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;</span></span></code></pre>
<p>If you would like to revert the bounded trait values, call
<code><a href="../reference/separate_trait_values.html">separate_trait_values()</a></code>:</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_trait_bound</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/separate_trait_values.html">separate_trait_values</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">austraits</span><span class="op">$</span><span class="va">definitions</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; # A tibble: 119 × 26</span></span>
<span><span class="co">#&gt;    dataset_id  taxon_name       observation_id trait_name value unit  entity_type value_type basis_of_value replicates</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;            &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span><span class="co">#&gt;  1 Cooper_2013 Acronychia baeu… 0071           fruit_len… 15    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  2 ABRS_2023   Acronychia aber… 01324          fruit_len… 16    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  3 ABRS_2023   Acronychia aber… 01324          fruit_len… 13    mm    species     minimum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  4 ABRS_2023   Acronychia eung… 01331          fruit_len… 12    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  5 ABRS_2023   Asterolasia ele… 02248          fruit_len… 10    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  6 ABRS_2023   Boronia angusti… 02910          fruit_len… 6     mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  7 ABRS_2023   Boronia quadril… 03056          fruit_len… 6     mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  8 ABRS_2023   Bosistoa floydii 03120          fruit_len… 10    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt;  9 ABRS_2023   Citrus australa… 04176          fruit_len… 50    mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; 10 ABRS_2023   Citrus garrawayi 04178          fruit_len… 100   mm    species     maximum    measurement    &lt;NA&gt;      </span></span>
<span><span class="co">#&gt; # ℹ 109 more rows</span></span>
<span><span class="co">#&gt; # ℹ 16 more variables: basis_of_record &lt;chr&gt;, life_stage &lt;chr&gt;, population_id &lt;chr&gt;, individual_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   repeat_measurements_id &lt;chr&gt;, temporal_context_id &lt;chr&gt;, source_id &lt;chr&gt;, location_id &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   entity_context_id &lt;chr&gt;, plot_context_id &lt;chr&gt;, treatment_context_id &lt;chr&gt;, collection_date &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   measurement_remarks &lt;chr&gt;, method_id &lt;chr&gt;, method_context_id &lt;chr&gt;, original_name &lt;chr&gt;</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Daniel Falster, Fonti Kar, Dony Indiarto.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
